addon/services/context-menu.js

```diff
@@ -5,21 +5,33 @@ import get from 'ember-metal/get';
 import set from 'ember-metal/set';
 import $   from 'jquery';
 
+const itemHeight    = 32;
+const safetyMarginX = 400;
+const safetyMarginY = 30;
+
 function renderLeft(xPosition, screenWidth) {
   if (!xPosition || !screenWidth) { return false; }
 
   let onRightHalf = xPosition > screenWidth * 0.5;
   let spaceRight  = screenWidth - xPosition;
 
-  return onRightHalf && spaceRight < 400;
+  return onRightHalf && spaceRight < safetyMarginX;
+}
+
+function correctedPositionY(yPosition, screenHeight, itemCount) {
+  let estimatedHeight = itemCount * itemHeight + safetyMarginY;
+  let breakPoint      = screenHeight - estimatedHeight;
+
+  return yPosition > breakPoint && itemCount ? breakPoint : yPosition;
```

Comment:
You don't have to check the itemCount, without items it won't render a context-menu and without itemCount your calculation for _estimatedHeight_ won't work either